src/graph.rs

```diff
@@ -460,6 +465,157 @@ where
 
         self.index = Some(index);
     }
+
+    // creates an AGraph from a list of 'good' addresses and
+    // the Edges array.
+    // For each edge, if both nodes are in the good list,
+    // a connection is created for each node.
+    // The AGraph has a equivalent information as the edges,
+    // but is just a node-centric view (required for closeness)
+    pub fn create_agraph(&self, addresses: &Vec<T>) -> AGraph {
+        let num_nodes = addresses.len();
+        let mut agraph: AGraph = AGraph::new();
+        for _ in 0..num_nodes {
+            agraph.push(Vertex::new());
+        }
+
+        // For all our edges, check if the nodes are in our address list
+        // We use the value of the addresses to find the index
+        // From then on, it's all integer indices for us
+        for edge in self.edges.iter() {
+            let source = *edge.source();
+            let target = *edge.target();
+
+            let src_result = addresses.iter().position(|&r| r == source);
```

Comment:
I _think_ you can use the already existing index in `Graph` to map addresses here, i.e. no need to create a new one?

Reply:
Possible, but here's maybe the rub... The `addresses` that get passed in are from the good nodes (should probably even be renamed `good_addresses`, which (if I'm not mistaken) are a subset of all the nodes in the graph.
```
{"jsonrpc":"2.0","result":{"num_known_nodes":13185,"num_good_nodes":3226,"num_known_connections":905381,
```

Reply:
And now that I think about it, the fact that I'm only analyzing the good nodes also has implications for using the adjacency matrix generated by Graph in the compute() function.  It would be done for **all** nodes, which would not only be computationally very difficult, but also something we probably don't really want.

Reply:
I see; the graph itself should be computing metrics on "good nodes" only as well. Let's keep these decoupled for now but we'll need to consider what distinction (if any) makes sense at the library level.

Reply:
(ah didn't see your second comment there before the refresh)